main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2351,8 +2352,17 @@ object Defaults extends BuildCommon {
         }
       )
     }
+    def onProgress(s: Setup) = {
+      val cp: BspCompileProgress = s.progress.asScala
+        .map(p => new BspCompileProgress(task, Some(p)))
+        .getOrElse(new BspCompileProgress(task, None))
```

Comment:
Can be simplified to
```suggestion
      val cp = new BspCompileProgress(task, s.progress.asScala)
```